### ISSUE TITLE
IKG regression in Flutter tools hot reload (empty kernel file)

### ISSUE THREAD

**Issue by aam** [dart-lang/sdk#31468]
Recent IKG change that required switch to minimal configuration in flutter tools regressed flutter hot reload.
`hotReloadVMReloadMillisecondsAfterChange` metric on [Flutter benchmark dashboard](https://flutter-dashboard.appspot.com/benchmarks.html) shows that hot reload after one line change effectively became a no-op.
The reason for that is that kernel file produced by IKG for that one line change ('Flutter Gallery'->'Updated Flutter Gallery' in app.dart) is effectively empty.

To reproduce it do `flutter run --preview-dart-2` from `examples/flutter_gallery`, edit lib/gallery/app.dart, reload flutter app (`r`) - observe no changes in the app, and lib/main.dart.dill file is empty(177 bytes, dumping it shows that it is empty)

Another way to reproduce the problem is with flutter engine frontend_server:
```
$ $FH/engine/src/out/host_debug_unopt/dart-sdk/bin/dart $FH/engine/src/out/host_debug_unopt/gen/frontend_server.dart.snapshot --sdk-root $FH/engine/src/out/android_debug_unopt/flutter_patched_sdk/ --no-link-platform --incremental --byte-store $FH/flutter/examples/flutter_gallery/build/ikg_byte_store
compile $FH/flutter/examples/flutter_gallery/lib/main.dart
result 1ff6ddd0-43c7-4299-901d-0aeb98aeecdf
Nit: line 1, column 235 of lib/gallery/item.dart: Import of 'Flow' (from 'package:flutter/src/widgets/basic.dart') hides import from 'dart:developer'.
Nit: line 1, column 229 of lib/demo/material/date_and_time_picker_demo.dart: Import of 'TextDirection' (from 'package:intl/intl.dart') hides import from 'dart:ui'.
Nit: line 1, column 271 of lib/demo/material/full_screen_dialog_demo.dart: Import of 'TextDirection' (from 'package:intl/intl.dart') hides import from 'dart:ui'.
Warning: line 1, column 11508 of ../../packages/flutter/lib/src/rendering/binding.dart: Superclass has no method named 'hitTest'.
Warning: line 1, column 11508 of ../../packages/flutter/lib/src/rendering/binding.dart: Superclass has no method named 'hitTest'.
1ff6ddd0-43c7-4299-901d-0aeb98aeecdf $FH/flutter/examples/flutter_gallery/lib/main.dart.dill
accept
recompile abc
$FH/flutter/examples/flutter_gallery/lib/gallery/app.dart
abc
result e68713d1-3fd1-45ca-93a7-4e6d0b294f1e
e68713d1-3fd1-45ca-93a7-4e6d0b294f1e $FH/flutter/examples/flutter_gallery/lib/main.dart.dill
quit
$ ll lib/main.dart.dill
-rw-r----- 1 user user 177 Nov 28 07:48 lib/main.dart.dill
```
cc @a-siva @sigmundch

**Comment by scheglov** [comment]
There are two CLs, either of which could have fixed the issue.
https://dart-review.googlesource.com/c/sdk/+/23462
https://dart-review.googlesource.com/c/sdk/+/15340

Currently the front-end team prefers avoid landing a workaround and wants to fix the issue in front-end code itself.

**Comment by scheglov** [comment]
The original issue for front-end to fix the problem with fileUri(s).
https://github.com/dart-lang/sdk/issues/31244

**Comment by sigmundch** [comment]
I'm closing to mark this as a duplicate of #31244

**Comment by sigmundch** [comment]
FWIW - the current workaround of using a different CWD in flutter tools is pretty tiny and I'm happy that this unblocks you @aam. As soon as #31244 gets fixed, we should delete that workaround.
